controllers/groupsController.js

```diff
@@ -29,61 +29,71 @@ const summary = async (req, res) => {
 
   try {
     const { uid } = req.params;
-    const postUpdateNewest = [];
-    const keywordListResult = await keywordModel.find({ ownerUid: uid }).exec();
-    const keywordIdList = keywordListResult.map((keyword) => keyword._id);
-    const postCreatedLast = [];
 
-    for await (const id of keywordIdList) {
-      const postResult = await postModel.find({ keywordId: id }).sort({ createdAt: -1 }).exec();
+    const keywordResult = await keywordModel.find({ ownerUid: uid }).sort({ updatedAt: -1 }).exec();
+    const keywordList = keywordResult.map((keyword) => {
+      return {
+        id: keyword._id.toString(),
+        name: keyword.keyword,
+      };
+    });
```

Comment:
```suggestion
    const keywordResult = await keywordModel
      .find({ ownerUid: uid }, { keyword: 1, updatedAt: 1 })
      .sort({ updatedAt: -1 })
      .exec();
```
map()을 사용해서 쿼리 결과를 다시 재조합하기보단 쿼리할 때 필요한 컬럼만 설정해서 가져오면 코드가 더 간결해질 것 같아요!
find()의 두번째 인자는 선택적으로 반환되길 원하는 컬럼만 설정할 수 있어 반환되길 원하는 컬럼을 1로 설정하면 되는데, 이렇게 변경하는 건 어떻게 생각하세요?!
[mongoose의 find() 공식문서](https://mongoosejs.com/docs/api/model.html#Model.find())

Reply:
덕분에 간결해진 쿼리로 유지보수가 편해졌습니다. 알지 못 했던 방법인데 알려주셔서 감사해요 한솔님!